### ISSUE TITLE
a11y issue: nested interative on tag that has onRemove and interactive/onClick

### ISSUE THREAD

**Issue by bvandrc** [palantir/blueprint#7552]
On a `Tag` that has either `interactive` or `onClick`, the outer `span` of the entire tag gets `role="button"`, but when `onRemove` is passed, a remove button is added inside of the tag. Having nested interactive elements is an accessibility failure, so this must be refactored to make the outer `button` the adjacent element to the `remove button`, not a parent of it.
